### PR TITLE
Add new agency setting and show agency settings in dashboard

### PR DESCRIPTION
**Pull request by terryttsai**
## Description of the change

Screen Shot 2023-01-06 at 4 03 16 PM
Screen Shot 2023-01-06 at 4 03 21 PM
Screen Shot 2023-01-06 at 4 03 25 PM

- Add new "Homepage URL" AgencySetting in the AgencySettings page
- Surface agency settings "HOMEPAGE_URL" and "PURPOSE_AND_FUNCTIONS" in agency dashboard.

## Type of change

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #XXXX

### Development

**This box MUST be checked by the submitter prior to merging**:
- [ ] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by lilidworkin** [comment]
@terryttsai  nice! I really like this solution! Would you mind just quickly discussing with @hobuobi to make sure that this addition to the Agency Settings page is okay?

File: agency-dashboard/src/Header/AboutModal.tsx [hunk 1]
```diff
@@ -48,12 +50,19 @@ export const AboutModal: React.FC<{
...
-            lectus lectus non sit justo. Massa, volutpat, diam maecenas risus.
-            Magna est neque tellus, ut in.
-          </AboutModalInfoBody>
-          <AboutModalButtonsContainer>
+          <AboutModalInfoBody>{agencyDescription}</AboutModalInfoBody>
+          <AboutModalButtonsContainer
+            onClick={() => {
+              if (agencyUrl) {
+                window.open(
+                  agencyUrl.match(/^http[s]?:\/\//)
```
Comment (lilidworkin) [inline agency-dashboard/src/Header/AboutModal.tsx:58]: smart!

File: agency-dashboard/src/Header/HeaderBar.tsx [hunk 2]
```diff
@@ -61,6 +61,16 @@ export const HeaderBar: React.FC<{ showTitle?: boolean }> = ({ showTitle }) => {
         <AboutModal
           closeModal={hideAboutModal}
           agencyName={agencyDataStore.agency?.name || ""}
+          agencyDescription={
+            agencyDataStore.agency?.settings.find(
```
Comment (lilidworkin) [inline agency-dashboard/src/Header/HeaderBar.tsx:65]: Would it make sense to move this logic to the agencyDataStore and create getters there that would make it easy to find settings of certain types, like agencyDataStore.getAgencyURL, etc?

File: publisher/src/components/Settings/AgencySettingsDescription.tsx [hunk 3]
```diff
@@ -63,8 +63,8 @@ export const AgencySettingsDescription: React.FC<{
   };
   const saveAgencyInfoChanges = () => {
     const updatedSettings = updateAgencySettings(
```
Comment (lilidworkin) [inline publisher/src/components/Settings/AgencySettingsDescription.tsx:65]: Would it be clearer to have specific setters for each kind of agency setting? Or better to have a general one like this? I actually don't have an opinion either way, could see an argument in both directions

**Comment by hobuobi** [comment]
Love it! Thanks for running with this :) Small nits/suggested improvements. 

### Design (cc @agronomic )
- P0: Use a single line text field rather than multi-line.
- P0: Need some padding underneath the text field to avoid it running into the buttons for save/cancel. 
- P0: The "read" view (not edit) should be an actual link.
- P1: Validation would be nice to verify that it's actually a URL. **If invalid:** 
    - Red treatment (similar to what we do for invalid data fields) 
    - Error message ("Please enter a valid URL.")
- P1: Empty state (will let Juan and Ivan handle this via #264).

### Other Copy
- Replace the "Type here..." with "URL of agency (e.g., https://doc.iowa.gov/)"
- Replace "Link to your agency's homepage." with "Provide a link to your agency's website."

**Comment by terryttsai** [comment]
Updated the PR to address the P0's. I'll leave URL validation alone for now, since there's so many different regex's out there to validate URLs and all of them cover a different subset of valid URLs (https://mathiasbynens.be/demo/url-regex)

**Comment by lilidworkin** [comment]
@terryttsai  is this still in draft or would you like a review?

File: publisher/src/components/Settings/AgencySettingsDescription.tsx (hunk 3 above)
Comment (terryttsai) [reply publisher/src/components/Settings/AgencySettingsDescription.tsx:60]: I do want to revise how the publisher saves and updates agency settings, and how the publisher fetches agency settings through the users endpoint. I'm thinking of doing this in a follow-up PR!

**Comment by terryttsai** [comment]
Followup: https://github.com/Recidiviz/justice-counts/issues/282

File: common/components/DataViz/utils.ts [hunk 4]
```diff
@@ -345,3 +345,7 @@ export const formatDateShort = (dateStr: string) => {
   const [, , month, year] = splitUtcString(dateStr);
   return `${abbreviatedMonths.indexOf(month) + 1}/${year}`;
 };
+
+export const formatExternalLink = (url: string) => {
```
[the remark this answers is not in the document]
Comment (mxosman) [reply common/components/DataViz/utils.ts:349]: I think this is a great solution!

**Pull request merged**